wgpu-types/src/lib.rs

```diff
@@ -569,6 +569,12 @@ bitflags::bitflags! {
         ///
         /// This is a native-only feature.
         const TEXTURE_COMPRESSION_ASTC_HDR = 1 << 43;
+        /// Skip buffer zero initialization
+        /// Supported Platforms:
+        /// - Metal
+        ///
+        /// This is a native-only feature.
+        const ALLOCATED_BUFFER_ALREADY_ZERO_FILLED = 1 << 44;
```

Comment:
I don't think this should end up in the public facing feature set here as this is something that the user doesn't need to know about at all. @kvark thoughts on where it should go instead?

Reply:
I think it's fine for some of the internal (i.e. needed on the hal - core boundary only) flags to be in this public bitflags list, as long as wgpu-core never reports them back to the user.
We could improve this by moving those internal flags into a separate bitflags type that would live in wgpu-hal.

Reply:
@kvark Not sure if I understand correctly, is the internal flags something like this?
```rust
bitflags::bitflags! {
	#[repr(transparent)]
	#[derive(Default)]
	pub struct InternalFeatures: u64 {
		const ALLOCATED_BUFFER_ALREADY_ZERO_FILLED = 1 << 0;
	}
}
// wgpu_hal/lib.rs
pub trait Device<A: Api>: Send + Sync {
	...
	fn contains_internal_feature(&self, _feature: wgt::InternalFeatures) -> bool;
}
// wgpu_hal/metal/mod.rs
pub struct Device {
	...
	internal_features: wgt::InternalFeatures,
}

// wgpu_hal/metal/device.rs
impl crate::Device<super::Api> for super::Device {
	...
	fn contains_internal_feature(&self, feature: wgt::InternalFeatures) -> bool {
		self.internal_features.contains(feature)
	}
}
```

Reply:
Let's discuss it as a follow-up, specifically aimed at the flags. Current PR should go in by just using the `Features` and having one there.